qiskit_experiments/library/randomized_benchmarking/interleaved_rb_experiment.py

```diff
@@ -88,56 +100,44 @@ def __init__(
-            element_lengths = [len(elements)] if self._full_sampling else lengths
-            std_circuits = self._generate_circuit(elements, element_lengths)
-            for circuit in std_circuits:
-                circuit.metadata["interleaved"] = False
-            circuits += std_circuits
-
-            int_elements = self._interleave(elements)
-            int_elements_lengths = [length * 2 for length in element_lengths]
-            int_circuits = self._generate_circuit(int_elements, int_elements_lengths)
-            for circuit in int_circuits:
-                circuit.metadata["interleaved"] = True
-                circuit.metadata["xval"] = circuit.metadata["xval"] // 2
-            circuits += int_circuits
-        return circuits
-
-    def _interleave(self, element_list: List) -> List:
-        """Interleaving the interleaved element inside the element list.
-
-        Args:
-            element_list: The list of elements we add the interleaved element to.
+    def circuits(self) -> List[QuantumCircuit]:
+        """Return a list of RB circuits.
 
         Returns:
-            The new list with the element interleaved.
-        """
-        new_element_list = []
-        for element in element_list:
-            new_element_list.append(element)
-            new_element_list.append(self._interleaved_element)
-        return new_element_list
-
-    def _set_interleaved_element(self, interleaved_element):
-        """Handle the various types of the interleaved element
-
-        Args:
-            interleaved_element: The element to interleave
-
-        Raises:
-            QiskitError: if there is no known conversion of interleaved_element
-            to a Clifford group element
+            A list of :class:`QuantumCircuit`.
         """
-        try:
-            interleaved_element_op = Clifford(interleaved_element)
-            self._interleaved_element = (interleaved_element, interleaved_element_op)
-        except QiskitError as error:
-            raise QiskitError(
-                "Interleaved element {} could not be converted to Clifford element".format(
-                    interleaved_element.name
-                )
-            ) from error
+        # Build circuits of reference sequences
+        reference_sequences = self._sample_sequences()
+        reference_circuits = self._sequences_to_circuits(reference_sequences)
+        for circ, seq in zip(reference_circuits, reference_sequences):
+            circ.metadata = {
+                "experiment_type": self._type,
+                "xval": len(seq),
+                "group": "Clifford",
+                "physical_qubits": self.physical_qubits,
+                "interleaved": False,
+            }
+        # Build circuits of interleaved sequences
+        interleaved_sequences = []
+        for seq in reference_sequences:
+            new_seq = []
+            for elem in seq:
```

Comment:
```python
interleaved_seq = np.insert(seq,  slice(1, len(seq)), self._interleaved_elem)
```
Just my preference.

Reply:
Such kind of simplification might be possible if all of elements are integers (it should be `np.insert(seq,  range(1, 1 + len(seq)), self._interleaved_elem)`), but the code above does not work since element may be Clifford. So let me leave these lines as they are.